README.md

```diff
@@ -345,6 +345,157 @@ api.get('/users', {}, { cancelToken: source.token })
+
+Two common patterns that work well:
+
+1. Use the native cookie jar (recommended)
+
+   ```js
+   import CookieManager from '@react-native-cookies/cookies'
+   import { create } from 'apisauce'
+
+   // Seed a cookie into the native jar (optional)
+   ;(async () => {
+     await CookieManager.set('https://example.com', {
+       name: 'session',
+       value: 'abc123',
+       domain: 'example.com',
+       path: '/',
+       secure: true,
+       httpOnly: true,
+     })
+   })()
+
+   const api = create({
+     baseURL: 'https://example.com',
+     withCredentials: true, // RN will add cookies from the native jar
+   })
+   ```
+
+2. Send a manual `Cookie` header for a call (opt out of the jar)
+
+   ```js
+   import { create } from 'apisauce'
+
+   const api = create({ baseURL: 'https://example.com' })
+   // Set Cookie per-request to avoid leaking it globally
+   api.get('/users', {}, { withCredentials: false, headers: { Cookie: 'session=abc123' } })
+   ```
+
+## Reading Set-Cookie from responses
+
+Servers can return multiple `Set-Cookie` headers. On React Native, the header values are sometimes
+flattened into a single comma‑separated string. Because cookie attributes like `Expires=...` also contain
+commas, splitting on `,` is unsafe.
+
+Use a tiny helper to split and parse safely, then (optionally) persist them to the native jar so future
+requests pick them up automatically:
+
+```js
+import { parse, splitCookiesString } from 'set-cookie-parser'
+import CookieManager from '@react-native-cookies/cookies'
+import { create } from 'apisauce'
+
+const api = create({ baseURL: 'https://example.com', withCredentials: true })
+
+api.addResponseTransform((res) => {
+  const headers = res.headers || {}
+  const raw = headers['set-cookie']
+    ?? headers['Set-Cookie']
+    ?? Object.entries(headers).find(([k]) => k.toLowerCase() === 'set-cookie')?.[1]
+  if (!raw) return
+
+  // RN may provide a string; this handles Expires=... commas correctly
+  const parts = Array.isArray(raw) ? raw : splitCookiesString(String(raw))
+  const cookies = parse(parts, { map: false })
+
+  // Persist to the native jar (optional). Fire-and-forget to avoid blocking.
+  const { url = '/', baseURL = '' } = res.config || {}
+  let origin = ''
+  try {
+    if (baseURL) {
+      origin = new URL(url || '', baseURL).origin
+    } else if (url) {
+      origin = new URL(String(url)).origin
+    }
+  } catch {
+    if (typeof baseURL === 'string') {
+      const m = baseURL.match(/^(https?:\/\/[^/]+)/i)
+      if (m) origin = m[1]
+    }
+  }
+  if (!origin || !/^https?:\/\//.test(origin)) return
+  // Compute a standards-compliant default path for new cookies
+  let defaultPath = '/'
+  try {
+    const reqUrl = baseURL ? new URL(url || '', baseURL) : new URL(String(url))
+    const p = reqUrl.pathname || '/'
+    defaultPath = p === '/' ? '/' : (p.endsWith('/') ? p : p.replace(/\/[^/]*$/, '/'))
+  } catch {}
+  Promise.all(
+    cookies.map((c) => {
```

Comment:
@CharlieHelps Let's extract the logic here in the map into a helper function, please. Provide good docs (JSDoc format) and excellent strict TypeScript types.